packages/shared/src/components/cards/list/ActionButtons.tsx

```diff
@@ -53,6 +53,15 @@ export default function ActionButtons({
     await onDownvoteClick?.(post);
   };
 
+  const onToggleUpvote = () => {
+    onUpvoteClick?.(post);
+    setUserUpvoted(true);
+
+    if (post.userState?.vote === UserVote.Down && !!showTagsPanel) {
+      onClose(true);
+    }
```

Comment:
Since `onUpvoteClick` is async it would always execute after this `if` but I think it makes more sense to move this part at the top of the function so you first check the current state and then execute upvote.

Reply:
Yeah, sure, we can do that.